internal/pkg/runtime/runtime.go

```diff
@@ -22,3 +27,41 @@ const (
 func (m Mode) String() string {
 	return [...]string{"Cloud", "Container", "Interactive", "Metal"}[m]
 }
+
+// Runtime defines the runtime parameters.
+type Runtime interface {
+	Mode() Mode
+	Platform() platform.Platform
+	Config() config.Configurator
+}
+
+// NewRuntime initializes and returns the runtime interface.
+func NewRuntime(m Mode, p platform.Platform, c config.Configurator) Runtime {
+	return &DefaultRuntime{
+		m: m,
+		p: p,
+		c: c,
+	}
+}
+
+// DefaultRuntime implements the Runtime interface.
+type DefaultRuntime struct {
```

Comment:
feel like this is another instance of naming is hard. `Default` doesn't seem like the right fit, but at the same time i don't think these implementation details leak through at all, so not sure it really matters much.

Reply:
Yep. Not sure what else to call it.